### PR TITLE
Improve the override deps docs

### PR DESCRIPTION
**Pull request by artembilan**

### CODE REVIEW

File: spring-kafka-docs/src/main/asciidoc/appendix.adoc [hunk 1]
```diff
@@ -56,37 +41,35 @@ When using 2.8.1, you also need to exclude `spring-kafka-test` 's `zookeeper` de
 
 <dependency>
     <groupId>org.apache.kafka</groupId>
-    <artifactId>kafka_2.13</artifactId>
```
Comment (garyrussell) [inline spring-kafka-docs/src/main/asciidoc/appendix.adoc:59]: Why is this removed?

File: spring-kafka-docs/src/main/asciidoc/appendix.adoc [hunk 2]
```diff
@@ -2,50 +2,35 @@
...
 
 <dependency>
-    <groupId>org.apache.kafka</groupId>
-    <artifactId>kafka-clients</artifactId>
-    <version>{kafka-version}</version>
-    <classifier>test</classifier>
+    <groupId>org.springframework.kafka</groupId>
+    <artifactId>spring-kafka-test</artifactId>
     <scope>test</scope>
-    <!-- needed if downgrading to Apache Kafka 2.8.1 -->
```
Comment (garyrussell) [inline spring-kafka-docs/src/main/asciidoc/appendix.adoc:48]: I see you have moved this to the right artifact, but why is the comment removed?

File: spring-kafka-docs/src/main/asciidoc/appendix.adoc [hunk 3]
```diff
@@ -56,37 +41,35 @@ When using 2.8.1, you also need to exclude `spring-kafka-test` 's `zookeeper` de
 
 <dependency>
     <groupId>org.apache.kafka</groupId>
-    <artifactId>kafka_2.13</artifactId>
-    <version>{kafka-version}</version>
+    <artifactId>kafka-clients</artifactId>
```
Comment (garyrussell) [inline spring-kafka-docs/src/main/asciidoc/appendix.adoc:44]: Why move the clients artifact down here? Either way, it should be adjacent to the kafka-streams artifact.

**Review by garyrussell — CHANGES_REQUESTED** [review]

File: spring-kafka-docs/src/main/asciidoc/appendix.adoc (hunk 3 above)
Comment (artembilan) [reply spring-kafka-docs/src/main/asciidoc/appendix.adoc:44]: This one is `<classifier>test</classifier>`. I don't see any relevance to the `kafka-streams`.

File: spring-kafka-docs/src/main/asciidoc/appendix.adoc [hunk 4]
```diff
@@ -56,37 +41,35 @@ When using 2.8.1, you also need to exclude `spring-kafka-test` 's `zookeeper` de
...
-    implementation 'org.springframework.kafka:spring-kafka:{project-version}'
-
-    implementation "org.apache.kafka:kafka-clients:$kafkaVersion"
-    implementation "org.apache.kafka:kafka-streams:$kafkaVersion" // optional - only needed when using kafka-streams
-    testImplementation ('org.springframework.kafka:spring-kafka-test:{project-version}') {
-        // needed if downgrading to Apache Kafka 2.8.1
-        exclude group: 'org.apache.zookeeper', module: 'zookeeper'
-    }
-    testImplementation "org.apache.kafka:kafka-clients:$kafkaVersion:test"
-    testImplementation "org.apache.kafka:kafka_2.13:$kafkaVersion"
```
[the remark this answers is not in the document]
Comment (artembilan) [reply spring-kafka-docs/src/main/asciidoc/appendix.adoc:87]: It is managed by Spring Boot.
Not sure how, but we don't need this dependency explicitly:
```
[INFO] +- org.springframework.kafka:spring-kafka-test:jar:2.8.0:test
[INFO] |  +- org.apache.kafka:kafka-metadata:jar:2.8.1:test
[INFO] |  +- org.apache.kafka:kafka-streams-test-utils:jar:2.8.1:test
[INFO] |  |  \- org.apache.kafka:kafka-streams:jar:2.8.1:test
[INFO] |  |     +- org.apache.kafka:connect-json:jar:2.8.1:test
[INFO] |  |     |  \- org.apache.kafka:connect-api:jar:2.8.1:test
[INFO] |  |     \- org.rocksdb:rocksdbjni:jar:5.18.4:test
[INFO] |  +- org.apache.kafka:kafka_2.13:jar:2.8.1:test
```

File: spring-kafka-docs/src/main/asciidoc/appendix.adoc (hunk 1 above)
Comment (artembilan) [reply spring-kafka-docs/src/main/asciidoc/appendix.adoc:59]: Same answer as for same dependency in Gradle config below.

Comment (garyrussell) [reply spring-kafka-docs/src/main/asciidoc/appendix.adoc:59]: Confused. Why do we still need `kafka-streams`? Boot manages that too.

Comment (garyrussell) [reply spring-kafka-docs/src/main/asciidoc/appendix.adoc:59]: spring-kafka too

Comment (garyrussell) [reply spring-kafka-docs/src/main/asciidoc/appendix.adoc:59]: Ignore me 😄

Comment (artembilan) [reply spring-kafka-docs/src/main/asciidoc/appendix.adoc:59]: Correct, it does manage them, but doesn't not provide.
We still have to specify artifacts in our poms, even if without versions.
The `<classifier>test</classifier>` are not managed by Boot, so we have to specify them explicitly since just plain version does not have an effect.

**Comment by garyrussell** [comment]
Added some doc polishing for your review.

**Comment by artembilan** [comment]
> Added some doc polishing for your review.

Looks good for me.
Thanks

**Review by garyrussell — APPROVED** [review]
